app/helpers/full_url_helper.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module FullUrlHelper
+  def url_helpers
+    # This is how we can get the helpers with a usable root_url outside the controllers
+    Rails.application.routes.default_url_options = ActionMailer::Base.default_url_options
+    Rails.application.routes.url_helpers
+  end
+
+  def full_checkout_path
+    URI.join(url_helpers.root_url, url_helpers.checkout_path).to_s
+  end
+
+  def full_order_path(order)
+    URI.join(url_helpers.root_url, url_helpers.order_path(order)).to_s
+  end
```

Comment:
IMO these two methods are redundant with https://guides.rubyonrails.org/routing.html#path-and-url-helpers.

I checked on a console and:

```rb
irb(main):001:0> Rails.application.routes.url_helpers.checkout_path
=> "/checkout"
irb(main):002:0> Rails.application.routes.url_helpers.checkout_url 
ArgumentError: Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true
        from /bundles/gems/actionpack-4.2.11.3/lib/action_dispatch/http/url.rb:45:in `full_url_for'
        from /bundles/gems/actionpack-4.2.11.3/lib/action_dispatch/http/url.rb:35:in `url_for'
        from /bundles/gems/actionpack-4.2.11.3/lib/action_dispatch/routing/route_set.rb:354:in `block in <class:RouteSet>'
        from /bundles/gems/actionpack-4.2.11.3/lib/action_dispatch/routing/route_set.rb:221:in `call'
        from /bundles/gems/actionpack-4.2.11.3/lib/action_dispatch/routing/route_set.rb:345:in `block (2 levels) in define_url_helper'
        from (irb):2
        from /bundles/gems/railties-4.2.11.3/lib/rails/commands/console.rb:110:in `start'
        from /bundles/gems/railties-4.2.11.3/lib/rails/commands/console.rb:9:in `start'
        from /bundles/gems/railties-4.2.11.3/lib/rails/commands/commands_tasks.rb:68:in `console'
        from /bundles/gems/railties-4.2.11.3/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
        from /bundles/gems/railties-4.2.11.3/lib/rails/commands.rb:17:in `<top (required)>'
        from bin/rails:4:in `require'
        from bin/rails:4:in `<main>'
irb(main):003:0> Rails.application.routes.default_url_options[:host]
=> nil
irb(main):004:0> Rails.application.routes.default_url_options[:host] = 'localhost:3000'
=> "localhost:3000"
irb(main):005:0> Rails.application.routes.url_helpers.checkout_url 
=> "http://localhost:3000/checkout"
```

By looking at how we do the same for the email config in https://github.com/openfoodfoundation/openfoodnetwork/blob/master/lib/spree/core/mail_settings.rb#L15 and checking the following on FR

```rb
irb(main):001:0> Spree::Config.site_url
=> "www.coopcircuits.fr"
```

looks like adding the following in `config/environments/*.rb` would be enough for Rails URL helpers to work

```rb
Rails.application.routes.default_url_options[:host] = Spree::Config.site_url
```

Reply:
I'm going to push this to tech debt in case there's unexpected stuff that comes up when messing with routes, so that we don't delay SCA getting out. https://github.com/openfoodfoundation/openfoodnetwork/issues/6860